### ISSUE TITLE
New version of ParametricLindenmayer.js using CsoundAC.PITV instead of ChordSpaceGroup

### ISSUE THREAD

**Issue by gogins** [gogins/csound-ac#21]
- [ ] Remove all references to Silencio's ChordSpaceGroup and replace them with CsoundAC's PITV class, which has a less buggy implementation of chord spaces.
- [ ] Test with a new version of an existing piece that uses ParametricLindenmayer.

**Comment by gogins** [comment]
The operations required include:

- [ ] clone, doesn't exist in CsoundAC, but _pretty much not needed I think_.
- [ ] new Chord is _OK_.
- [ ] new Event is _OK_.
- [ ] numeric.add looks problematic.
- [ ] numeric.mul looks problematic.
- [ ] Event fields in JavaScript are properties, _these don't exist in CsoundAC_.
- [ ] Eigen certainly can be compiled in embound, but _creating a JavaScript binding that exposes Eigen objects in their full glory is not trivial_, see https://observablehq.com/@rreusser/eigen.

Bottom line: I am not doing this now.

In reality, this is probably not happening unlss and until a real C++ compiler can be embedded in a cloud-5 piece.

**Comment by gogins** [comment]
Well, if I do do it, I need to expose some basic linear algebra operations implemented using Eigen in CsoundAC. That would include all basic operations between scalars, vectors, and matrices (add, subtract, multiply, divide). Because there is no type overloading in JavaScript, that would be something like (keeping in mind, in Eigen vectors are column or row matrices), where X is the operation such as add, sub, mul, div:
```
Xsme(scalar, matrix, index)
Xsm(scalar, matrix)
Xmm(matrix, matrix)
```



```

**Comment by gogins** [comment]
Also I need to look at the state of the PITV implementation in csound-wasm.

**Comment by gogins** [comment]
add
subtract
multiply
divide
scalar_multiply
scalar_divide

**Comment by gogins** [comment]
This is all doable, but it's a lot of work. I am putting this off till after I finish my piece for the NYCEMF.

**Comment by gogins** [comment]
I'm not sure I need the new matrix arithmetic methods, but I may implement them anyway.

I've made a good start...
